model/histogram/float_histogram.go

```diff
@@ -504,6 +599,10 @@ func (h *FloatHistogram) DetectReset(previous *FloatHistogram) bool {
 	if h.Count < previous.Count {
 		return true
 	}
+	if h.HasCustomBounds() != previous.HasCustomBounds() {
+		// TODO(zenador): correct?
+		return true
+	}
 	if h.Schema > previous.Schema {
```

Comment:
If both histograms have custom buckets, we need to check the bucket definitions, if they are different then that can only happen by restart of the application (no dynamic buckets) so it's a reset.